pkg/slurp/slurp.go

```diff
@@ -315,7 +315,12 @@ func (s Slurper) GetSecretsAsync(detectrs ...detectors.Detector) (chan SecretRes
 			var err error
 			keywords := detector.Keywords()
 			for _, keyword := range keywords {
-				messageChan, err2Chan := s.SearchMessagesAsync(fmt.Sprintf("%s*", keyword))
+				var searchString = keyword
+				var re = regexp.MustCompile(`(-|\.|_)$`)
+				if !re.MatchString(keyword) {
+					searchString = searchString + "*"
+				}
+				messageChan, err2Chan := s.SearchMessagesAsync(fmt.Sprintf("%s", searchString))
```

Comment:
I am a little picky when it comes to the style so I apologize. The following is how I would do this, the first line would be on line 314:
```go
nonStarSearchableRe := regexp.MustCompile(`(-|\.|_)$`)
for _, detector := range selectedDetectors {
	var err error
	keywords := detector.Keywords()
	for _, keyword := range keywords {
		if !nonStarSearchableRe.MatchString(keyword) {
			keyword = keyword + "*"
		}
		messageChan, err2Chan := s.SearchMessagesAsync(keyword)
```

I know that technically `searchString` better describes what is being passed to `s.SearchMessagesAsync()` but I don't think we need to create a brand new variable just for that.